src/GraphQL/Language/AST/Document.cs

```diff
@@ -26,13 +23,38 @@ public Document()
         public string OriginalQuery { get; set; }
 
         /// <inheritdoc/>
-        public override IEnumerable<INode> Children => _definitions;
+        public override IEnumerable<INode> Children
+        {
+            get
+            {
+                if (Operations.List != null)
+                {
+                    foreach (var o in Operations.List)
+                        yield return o;
```

Comment:
Are you sure `yield return` doesn't create an allocation?  Pretty sure it does, but maybe not.  Is `Children` used on the hot path?  If not, this would be better anyway.

Reply:
No, `Children` is not used anymore on hot path, only in some places inside `ToString()` for printing or debug purposes.